src/Metric.jl

```diff
@@ -1,50 +1,202 @@
+@doc doc"""
+    Metric
+
+Abstract type for the pseudo-Riemannian metric tensor $g$, a family of smoothly
+    varying inner products on the tangent space. See [`inner`](@ref).
+"""
 abstract type Metric end
 
+@doc doc"""
+    RiemannianMetric <: Metric
+
+Abstract type for Riemannian metrics, a family of positive definite inner
+    products. The positive definite property means that for $v \in T_x M$,
+    the inner product $g(v,v) > 0$.
```

Comment:
For `x` different from 0.

Reply:
Do you mean `v` different from the zero vector? I think so. It should be $g(v,v) \geq 0$ for $v \in T_x M$ and $g(v,v) = 0$ iff $v=0$,

Reply:
Yes, of course, `v` is different from the zero vector.